### PR TITLE
GEN-664 - refact(ComparisonTableBlock): makes usage of Mobile/Desktop comparison table components

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Updates `<ComparisonTableBlock />` (CMS) so it renders the correct version of _Comparison Table_ based on viewport size. Please don't mind the '*' that get's shown before each table attribute. It's a side effect of how _Desktop_  Comparison Table get's built. I gonna need to change cat/dog insurance pages a little bit when this get's released so maybe it would be better to feature flag this? 🤔  

## Justify why they are needed

We got a new [design](https://www.figma.com/file/qUhLjrKl98PAzHov9ilaDH/Hedvig-UI-Kit?type=design&node-id=3380-13894&mode=design&t=snZZXMBD7zxyZULO-4) for a _mobile_ version of the comparison table:
image

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2748** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2748" target="_blank">Graphite</a> 
    * **PR #2749** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2749" target="_blank">Graphite</a> 
      * **PR #2750** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2750" target="_blank">Graphite</a> 
        * **PR #2751** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2751" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2751?utm_source=stack-comment).

File: apps/store/src/blocks/ComparisonTableBlock.tsx [hunk 1]
```diff
@@ -13,49 +17,27 @@ type Props = SbBaseBlockProps<{
...
-                      {getCellValue(cell.value)}
-                    </ComparisonTable.DataCell>
-                  ),
-                )}
-              </ComparisonTable.Row>
-            ))}
-          </ComparisonTable.Body>
-        </ComparisonTable.Root>
+        {!matchesMdAndUp && <MobileComparisonTable {...table} {...storyblokEditable(blok)} />}
+        {matchesMdAndUp && <DesktopComparisonTable {...table} {...storyblokEditable(blok)} />}
```
Comment (robinandeer) [inline apps/store/src/blocks/ComparisonTableBlock.tsx:37]: normal terneries are ok, right? might look more idiomatic React

```
{matchesMdAndUp ? (
  <DesktopComparisonTable {...table} {...storyblokEditable(blok)} />
) : (
  <MobileComparisonTable {...table} {...storyblokEditable(blok)} />
)
```

or even:

```
const ComparisonTable = matchesMdAndUp ? DesktopComparisonTable : MobileComparisonTable

<ComparisonTable {...table} {...storyblokEditable(blok)} />
```

File: apps/store/src/blocks/ComparisonTableBlock.tsx [hunk 2]
```diff
@@ -13,49 +17,27 @@ type Props = SbBaseBlockProps<{
...
       </GridLayout.Content>
     </GridLayout.Root>
   )
 }
 
-const getCellValue = (value: string) => {
-  if (value === '[*]') return <ComparisonTable.CheckIcon />
-  if (value === '[]') return <ComparisonTable.MissingIcon />
-  return value
-}
```
Comment (robinandeer) [inline apps/store/src/blocks/ComparisonTableBlock.tsx:59]: I still think this was kinda smart but maybe a bit over-generalizing 😅

**Review by robinandeer — APPROVED** [review]
Nice work

**Comment by guilhermespopolin** [comment]
@guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2751).

**Comment by guilhermespopolin** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2751) rebased this pull request as part of a merge.

**Comment by guilhermespopolin** [comment]
@guilhermespopolin merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2751).

**Pull request merged**